code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -954,6 +954,11 @@
 	gib()
 	return
 
+/mob/living/silicon/robot/update_canmove() // to fix lockdown issues w/ chairs
+    if(lockcharge)
+        canmove = 0

```

Comment:
Then this current form doesn't work either. Because that line 819 will override absoloutely everything you do here, unless you put in a `return` directive in this branch. But that'll cause even more fuckups, as `update_canmove` handles more than just canmove variable.

Now, the easiest method of doing this is:

```
/mob/living/silicon/robot/update_canmove()
	. = ..()
	if (lockcharge)
		. = 0
```